python/pyspark/sql/connect/client.py

```diff
@@ -867,6 +878,8 @@ def _analyze(self, method: str, **kwargs: Any) -> AnalyzeResult:
                 req.unpersist.blocking = cast(bool, kwargs.get("blocking"))
         elif method == "get_storage_level":
             req.get_storage_level.relation.CopyFrom(cast(pb2.Relation, kwargs.get("relation")))
+        elif method == "resources":
```

Comment:
I'm not sure this is a good approach as it overloads the semantics of analyze with generic system metadata information. 

My suggestion is to implement resource metadata using a command instead.

Reply:
I feel `sc.resources` itself is not like a command, but similar to `spark.version` which also uses the analyze rpc

Reply:
It's the same issue. I don't appreciate the way it was implemented. The analyze RPC has no semantic relationship to either version nor resources. Please change the resources now as it's too late for version.